packages/teams-js/src/public/video.ts

```diff
@@ -88,11 +88,33 @@ export namespace video {
     notifyError: (errorMessage: string) => void,
   ) => void;
 
+  /**
+   * Predefined failure reasons for preparing the selected video effect
+   * @beta
+   */
+  export enum EffectFailureReason {
+    /**
+     * A wrong effect id is provided
+     */
+    InvalidEffectId = 'InvalidEffectId',
+    /**
+     * The effect can't be initialezed
+     */
+    InitializationFailure = 'InitializationFailure',
+  }
+
+  enum EffectSuccessReason {
+    EffectPrepared = 'EffectPrepared',
+
+    NoPromiseReturned = 'NoPromiseReturned',
+  }
+
   /**
    * Video effect change call back function definition
+   * Return a Promise which will be resolved when the effect is prepared, or reject with a failure reason
    * @beta
    */
-  export type VideoEffectCallBack = (effectId: string | undefined) => void;
+  export type VideoEffectCallBack = (effectId: string | undefined) => Promise<void> | void;
```

Comment:
>VideoEffectCallBack [](http://example.com/codeflow?start=14&length=19)

Everywhere else in here treats callback like one word, which I think is the most common usage. As such, I recommend this be `VideoEffectCallback` (lower-case b) and not `VideoEffectCallBack`

Reply:
updated